### PR TITLE
Feature/8425 launch store screen

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: #8425

### Description

This PR implements launch your store task according to design:

Screenshot 2023-03-10 at 13 06 24

This PR doesn't include the following: 
- Error handling. Is done here: #8528
- Using the right authenticated web view to be able to show the preview correctly when the store is not launched yet. @0nko was taking a look into that. More context here: pe5sF9-15a-p2#comment-1828
- Refreshing the current plan state after upgrading to the `eCommerce` plan from free trial. That should be part of Eagles's work. 
- Refresh the onboarding list state after store is launched. This should be tackled in a different PR where we add the new refresh policy

### Testing instructions

**eCommerce** plan test
- Create a new store. Preferably using store creation flow from mobile. Disable IAP if needed by updating `IsIAPEnabled` class
- Click on launch store task from onboarding list
- When store launched successfully check the preview is displayed correctly
- Click on share url and check Android share bottom sheet is displayed
- Click on navigate back to store to land on My Store tab again. 

Free trial plan test
- Create a new free trial site by opening this endpoint `https://wordpress.com/setup/wooexpress`
- Click on launch store task
- Check launch task is disable
- Click on the banner and check you are redirected to plans web view for a plan upgrade. 

### Images/gif

Free Trial

https://user-images.githubusercontent.com/2663464/224321176-19180344-517e-46eb-9210-be29bff0062b.mp4

Publish store

https://user-images.githubusercontent.com/2663464/224321202-0b09aae4-c71d-47c3-8bf0-60e9b7d99567.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingViewModel.kt [hunk 1]
```diff
@@ -1,21 +1,26 @@
 package com.woocommerce.android.ui.login.storecreation.onboarding
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingViewModel.kt:1]: Changes in this file are unrelated to the Launch Store task handling. I just did a small refactor to remove code duplication and make everything easier to handle clicks for any of the onboarding tasks.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/SiteModelExt.kt [hunk 2]
```diff
@@ -39,3 +39,7 @@ val SiteModel.isSimpleWPComSite
 
 val SiteModel.adminUrlOrDefault
     get() = adminUrl ?: url.slashJoin("wp-admin")
+
+const val ECOMMERCE_TRIAL_PLAN_ID = 1052L
+val SiteModel.isCurrentPlanEcommerceTrial
+    get() = planId == ECOMMERCE_TRIAL_PLAN_ID
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/SiteModelExt.kt:64]: Very likely this will be replaced by the same extension already added in [this PR](https://github.com/woocommerce/woocommerce-android/pull/8379#discussion_r1130818382). Will depend on which one is merged first

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/Tag.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.text.font.FontWeight
+import com.woocommerce.android.R
+
+@Composable
+fun WcTag(
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/Tag.kt:18]: This tag can be re-used in `DomainDashboardScreen` to replace `PrimaryDomainTag`. I have not added that change here because I did not want to add unnecessary changes, but should be a pretty straightforward change.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingScreen.kt [hunk 4]
```diff
@@ -51,7 +51,11 @@ import androidx.compose.ui.unit.DpOffset
 import androidx.compose.ui.unit.dp
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/onboarding/StoreOnboardingScreen.kt:51]: Changes added to this file are related to making the onboarding tasks "clickable" with the proper ripple effect.

**Review by 0nko — APPROVED** [review]
Excellent work! 🎖️ 

The code looks good and the store launching behaves as expected 👍. The launched site preview doesn't look that great but I've refactored the `WCWebView` in a different PR that will allow us to fix that.

:shipit:
